server.js

```diff
@@ -0,0 +1,247 @@
+      return res.end(`<html><body>Not Authorized</body></html>`);
+    }
+  }
+  // DELETE
+
+  if (req.method === 'DELETE') {
+    if (req.headers.authorization) {
+      const encodedString = req.headers.authorization.slice(6);
+      const base64Buffer = new Buffer(encodedString, 'base64');
+      const decodedString = base64Buffer.toString();
+      let username = decodedString.slice(0, decodedString.indexOf(':'));
+      let password = decodedString.slice(decodedString.indexOf(':') + 1);
+
+      if (authorizedUsers.hasOwnProperty(username) && authorizedUsers[username] === password) {
+        let dontDeleteFiles = ['/server.js', '/elementtest.js', '/css', '/.keep'];
+
+        if (!dontDeleteFiles.includes(req.url)) {
+          fs.unlink(`./public${req.url}`, (err) => {
+            if (err) throw err;
+            // update index
+            updateIndex();
+            return res.end(`${req.url} was deleted  `);
+          });
+        }
+      } else {
+        res.writeHead(401, { 'WWW-Authenticate': 'Basic realm="SecureArea"' });
+        return res.end(`<html><body>Invalid Authentication Credentials</body></html>`);
+      }
+    } else {
+      res.writeHeader(401, { 'WWW-Authenticate': 'Basic realm="SecureArea"' });
+      return res.end(`<html><body>Not Authorized</body></html>`);
+    }
+  }
+});
+
+server.listen(8080, () => {
+  console.log(`Server is ON`);
+});
+
+// updateIndex function
+function updateIndex() {
+  fs.readdir(`./public`, (err, files) => {
+    if (err) {
+      throw new Error(err);
+    } else {
+      let numberOfElements = 0;
+      let ignoreFiles = ['server.js', 'elementtest.js', 'css', '.keep'];
+      let fileList = '';
+      files.forEach((file) => {
+        if (!ignoreFiles.includes(file)) {
+          numberOfElements++;
+          let fileName = file.slice(0, file.indexOf('.'));
+          fileList += `
+          <li>
+          <a href="${file}">${fileName}</a>
+        </li>`;
+        }
+      });
+      let indexTemplate = `
+      <!DOCTYPE html>
+<html lang="en">
+<head>
+<meta charset="UTF-8">
+<title>The Elements</title>
+<link rel="stylesheet" href="/css/styles.css">
+</head>
+<body>
+<h1>The Elements</h1>
+<h2>These are all the known elements.</h2>
+<h3>These are ${numberOfElements}<!DOCTYPE html>
+<html lang="en">
+<head>
+<meta charset="UTF-8" />
+<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+<meta http-equiv="X-UA-Compatible" content="ie=edge" />
+<title>Document</title>
+<link rel="stylesheet" href="css/styles.css" />
+</head>    const decodedString = base64Buffer.toString();
+let username = decodedString.slice(0, decodedString.indexOf(':'));
+let password = decodedString.slice(decodedString.indexOf(':'));
+if (authorizedUsers.hasOwnProperty(username) && authorizedUsers[username] === password) {
+  postOrPut();
+<body></body>
+</html></h3>
+<ol>
+${fileList}
+</ol>
+</body>
+</html>`;
+    }
+  });
```

Comment:
It looks like this creates the proper index.html file, but you don't write it out to your public dir.